docs/migrations/chain-upgrade-guide-043.md

```diff
@@ -0,0 +1,233 @@
+```
+
+Now that `cosmovisor` is installed and the genesis binary has been added, let's add the upgrade handler to `simapp/app.go` and prepare the upgrade binary.
+
+### Chain Upgrade
+
+<!-- TODO: update example to use v0.43.0 -->
+
+Check out `release/v0.43.x`:
+
+```
+git checkout release/v0.43.x
+```
+
+Add the following to `simapp/app.go` starting on line 260:
+
+```go
+	app.registerUpgradeHandlers()
+```
+
+Add the following to `simapp/app.go` starting on line 420 (to learn more about the upgrade handler, see the [In-Place Store Migrations](../core/upgrade.html)):
+
+```go
+func (app *SimApp) registerUpgradeHandlers() {
+	app.UpgradeKeeper.SetUpgradeHandler("v0.43", func(ctx sdk.Context, plan upgradetypes.Plan, _ module.VersionMap) (module.VersionMap, error) {
+		// 1st-time running in-store migrations, using 1 as fromVersion to
+		// avoid running InitGenesis.
+		fromVM := map[string]uint64{
+			"auth":         1,
+			"bank":         1,
+			"capability":   1,
+			"crisis":       1,
+			"distribution": 1,
+			"evidence":     1,
+			"gov":          1,
+			"mint":         1,
+			"params":       1,
+			"slashing":     1,
+			"staking":      1,
+			"upgrade":      1,
+			"vesting":      1,
+			"ibc":          1,
+			"genutil":      1,
+			"transfer":     1,
+		}
+
+		return app.mm.RunMigrations(ctx, app.configurator, fromVM)
+	})
+
+	upgradeInfo, err := app.UpgradeKeeper.ReadUpgradeInfoFromDisk()
+	if err != nil {
+		panic(err)
+	}
+
+	if upgradeInfo.Name == "v0.43" && !app.UpgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
+		storeUpgrades := storetypes.StoreUpgrades{
+			Added: []string{"authz", "feegrant"},
+		}
+
+		// configure store loader that checks if version == upgradeHeight and applies store upgrades
+		app.SetStoreLoader(upgradetypes.UpgradeStoreLoader(upgradeInfo.Height, &storeUpgrades))
+	}
+}
+```
+
+Add `storetypes` to imports:
+
+```go
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+```
+
+Build the `simd` binary for `v0.43.0` (the upgrade binary):
+
+```
+make build
+```
+
+Create the folder for the upgrade binary and copy the `v0.43.0` binary:
+
+```
+mkdir -p $DAEMON_HOME/cosmovisor/upgrades/v0.43/bin
+cp ./build/simd $DAEMON_HOME/cosmovisor/upgrades/v0.43/bin
+```
+
+Now that we have added the upgrade handler and prepared the upgrade binary, we are ready to start `cosmovisor` and simulate the upgrade proposal process.
+
+### Upgrade Proposal
+
+Start the node using `cosmovisor`:
+
+```
+cosmovisor start
+```
+
+Open a new terminal window and submit an upgrade proposal along with a deposit and a vote (these commands must be run within 20 seconds of each other):
+
+```
+./build/simd tx gov submit-proposal software-upgrade v0.43 --title upgrade --description upgrade --upgrade-height 20 --from validator --yes
+./build/simd tx gov deposit 1 10000000stake --from validator --yes
+./build/simd tx gov vote 1 yes --from validator --yes
+```
+
+Confirm the chain automatically upgrades at height 20.
```

Comment:
I believe we should add a paragraph about migrating app.toml. i think there's only the rosetta section that's new

Reply:
Added a paragraph about two new sections (Rosetta and gRPC Web) and new option added to State Sync [here](https://github.com/cosmos/cosmos-sdk/pull/9567/files#diff-14b0b84dc6f1242582f3d0500e4285c9797aa72f08cedd8102ab51e7f8f784c6R37-R39 ).